### PR TITLE
Remove salt length check from Rfc2898DeriveBytes

### LINKED ISSUE #56908: Consider removing minimum salt length requirement from Rfc2898DerivedBytes
The instance-based PBKDF2 implementation in `Rfc2898DerivedBytes` currently throws an `ArgumentException` if a user supplied salt is less than 8 bytes.

https://github.com/dotnet/runtime/blob/1a4675db08384e4f20cf4481e6eaf03e40f3d754/src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.cs#L17

I would propose to remove this limitation. Though I do not know the origin of this requirement, one can reasonably assume that it was added to help developers avoid making a mistake of providing an inadequate salt.

However, I think there are some issues surrounding this.

1. As far as minimums go, 8 (64-bits) is too low in the first place. NIST SP800-132 says it should be 16 bytes (so 128 bits). Following the recommendations of appendix A.2.1, even larger is often used and suggested. Raising the minimum would probably be a breaking change that has no reasonable work arounds, so 8 seems at best arbitrary, at worse insufficient.
2. The minimum may be mistakenly thought of as "safe". Since the implementation has a minimum, it's not a stretch to interpret that as a recommendation or at least a safety boundary (which as described above, is not suitable in 2021). This should be handled in documentation, not in code.
3. The static one-shot `Pbkdf2` that was recently added has no requirement on the minimum salt size.
4. The minimum forces developers to come up with alternative solutions when they are required to use a salt that is out of their control. aspnet had support for arbitrary salt sizes and had work arounds to support salt lengths less than 8. This was removed when it was switched to the one-shot, but shows evidence that some are working around the requirement.

### PR DESCRIPTION
**Pull request by vcsjones**
Since the idea to remove the salt length check in #56908 received positive signals, I thought I would throw up a PR and see what happens.

Closes #56908.

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs [hunk 1]
```diff
@@ -346,7 +329,6 @@ public static void GetBytes_KnownValues_4()
                 });
         }
 
-#if NETCOREAPP
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:349]: Unrelated test cleanup: these tests don't build for anything other the `NETCOREAPP`, so the `#if` is not needed.

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs [hunk 2]
```diff
@@ -503,6 +484,16 @@ private static IEnumerable<KnownValuesTestCase> GetKnownValuesTestCases()
                 AnswerHex = "3D2EEC4FE41C849B80C8D83662C0E44A8B291A964CF2F07038",
             };
 
+            yield return new KnownValuesTestCase
+            {
+                CaseName = "RFC 6070 Case 1",
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:489]: My obsessive nature wants this above 6070#5 (the previous case), but if this is the only reason to spin the build I wouldn't bother.

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs [hunk 3]
```diff
@@ -173,13 +155,14 @@ public static void MinimumAcceptableInputs()
         {
             byte[] output;
 
-            using (var deriveBytes = new Rfc2898DeriveBytes("", new byte[8], 1))
+            using (var deriveBytes = new Rfc2898DeriveBytes("", new byte[0], 1))
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:158]: Could be Array.Empty, but it's test code so not really important.

File: src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs (hunk 2 above)
Comment (vcsjones) [reply src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898Tests.cs:489]: I looked through my PR again and realized I never removed to localized exception, so, let's use that as a reason to push and fix the nits while we are at it.

**Comment by bartonjs** [comment]
Opened https://github.com/dotnet/dotnet-api-docs/issues/6996 to track that the docs update for this is... not obvious to me.  :smile:

**Review by bartonjs — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
